client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/pattern-selector.tsx

```diff
@@ -0,0 +1,101 @@
+import { Button } from '@automattic/components';
+import classNames from 'classnames';
+import { useTranslate } from 'i18n-calypso';
+import { useState, useEffect } from 'react';
+import { getPatternPreviewUrl, handleKeyboard } from './utils';
+import type { Pattern } from './types';
+
+type PatternsSelectorProps = {
+	patterns: Pattern[] | null;
+	onSelect: ( selectedPattern: Pattern | null ) => void;
+	onDeselect: ( selectedPattern: Pattern | null ) => void;
+	title: string | null;
+	pattern: Pattern | null;
+	show: boolean;
+};
+
+const PatternsSelector = ( {
+	patterns,
+	onSelect,
+	onDeselect,
+	title,
+	pattern,
+	show,
+}: PatternsSelectorProps ) => {
+	const [ selectedPattern, setSelectedPattern ] = useState< Pattern | null >( null );
```

Comment:
According to p1662007297464909/1661949589.098779-slack-CRWCHQGUB, we could simplify this component 🙂

More context: p1662013741257909-slack-CRWCHQGUBI